# Analyse du marché Airbnb à New York (2024)
## Identification d'opportunités d'investissement immobilier

**Contexte :** Étude pour un fonds d'investissement immobilier visant à identifier les meilleures opportunités sur le marché de la location courte durée à New York.

**Source des données :** [New York Airbnb Open Data 2024 - Kaggle](https://www.kaggle.com/datasets/vrindakallu/new-york-dataset)

---

## Étape 1 — Sourcing et compréhension de la donnée

In [1]:
# Bibliothèques de manipulation de données
import pandas as pd
import numpy as np

# Bibliothèques de visualisation
import matplotlib.pyplot as plt
import seaborn as sns

# Configuration d'affichage
pd.set_option('display.max_columns', None)   # Voir toutes les colonnes
pd.set_option('display.float_format', '{:.2f}'.format)  # 2 décimales pour les floats
sns.set_style("whitegrid")                   # Style des graphiques seaborn
plt.rcParams['figure.figsize'] = (10, 6)     # Taille par défaut des graphiques

In [2]:
# Chargement du jeu de données brut
df = pd.read_csv('../data/raw/new_york_listings_2024.csv')

# Aperçu rapide : nombre de lignes et de colonnes
print(f"Dimensions du dataset : {df.shape[0]:,} lignes × {df.shape[1]} colonnes")
df.head()

Dimensions du dataset : 20,758 lignes × 22 colonnes


,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license,rating,bedrooms,beds,baths
0,1312228,Rental unit in Brooklyn · ★5.0 · 1 bedroom,7130382,Walter,Brooklyn,Clinton Hill,40.68,-73.96,Private room,55.00,30,3,2015-12-20,0.03,1,0,0,No License,5.0,1,1,Not specified
1,45277537,Rental unit in New York · ★4.67 · 2 bedrooms ·...,51501835,Jeniffer,Manhattan,Hell's Kitchen,40.77,-73.99,Entire home/apt,144.00,30,9,2023-05-01,0.24,139,364,2,No License,4.67,2,1,1
2,971353993633883038,Rental unit in New York · ★4.17 · 1 bedroom · ...,528871354,Joshua,Manhattan,Chelsea,40.75,-73.99,Entire home/apt,187.00,2,6,2023-12-18,1.67,1,343,6,Exempt,4.17,1,2,1
3,3857863,Rental unit in New York · ★4.64 · 1 bedroom · ...,19902271,John And Catherine,Manhattan,Washington Heights,40.84,-73.94,Private room,120.00,30,156,2023-09-17,1.38,2,363,12,No License,4.64,1,1,1
4,40896611,Condo in New York · ★4.91 · Studio · 1 bed · 1...,61391963,Stay With Vibe,Manhattan,Murray Hill,40.75,-73.98,Entire home/apt,85.00,30,11,2023-12-03,0.24,133,335,3,No License,4.91,Studio,1,1


### 1.1 Informations générales sur le dataset

In [3]:
# Types de données et valeurs non-nulles par colonne
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 20758 entries, 0 to 20757
Data columns (total 22 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              20758 non-null  int64  
 1   name                            20758 non-null  str    
 2   host_id                         20758 non-null  int64  
 3   host_name                       20758 non-null  str    
 4   neighbourhood_group             20758 non-null  str    
 5   neighbourhood                   20758 non-null  str    
 6   latitude                        20758 non-null  float64
 7   longitude                       20758 non-null  float64
 8   room_type                       20758 non-null  str    
 9   price                           20758 non-null  float64
 10  minimum_nights                  20758 non-null  int64  
 11  number_of_reviews               20758 non-null  int64  
 12  last_review                     20758 non-n

### 1.2 Statistiques descriptives

In [4]:
# Statistiques descriptives des colonnes numériques
df.describe()

,id,host_id,latitude,longitude,price,minimum_nights,number_of_reviews,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,beds
count,20758.00,20758.00,20758.00,20758.00,20758.00,20758.00,20758.00,20758.00,20758.00,20758.00,20758.00,20758.00
mean,303404386773971136.00,174931562.15,40.73,-73.94,187.78,28.56,42.64,1.26,18.84,205.99,10.85,1.72
std,390121559478713024.00,172554113.31,0.06,0.06,1022.80,33.54,73.56,1.90,70.91,135.09,21.36,1.21
min,2595.00,1678.00,40.50,-74.25,10.00,1.00,1.00,0.01,1.00,0.00,0.00,1.00
25%,27088080.00,20417384.50,40.68,-73.98,80.00,30.00,4.00,0.21,1.00,87.00,1.00,1.00
50%,49930032.50,108727121.00,40.72,-73.95,125.00,30.00,14.00,0.65,2.00,215.00,3.00,1.00
75%,721601888994556672.00,314410176.00,40.76,-73.92,199.00,30.00,49.00,1.80,5.00,353.00,15.00,2.00
max,1054375518474819584.00,550403525.00,40.91,-73.71,100000.00,1250.00,1865.00,75.49,713.00,365.00,1075.00,42.00


In [5]:
# Comptage des valeurs manquantes par colonne, triées
missing = df.isnull().sum().sort_values(ascending=False)
missing_pct = (missing / len(df) * 100).round(2)

pd.DataFrame({
    'Valeurs manquantes': missing,
    'Pourcentage (%)': missing_pct
})

,Valeurs manquantes,Pourcentage (%)
id,0,0.00
name,0,0.00
host_id,0,0.00
host_name,0,0.00
neighbourhood_group,0,0.00
neighbourhood,0,0.00
latitude,0,0.00
longitude,0,0.00
room_type,0,0.00
price,0,0.00


---
## Étape 2 — Nettoyage des données

### 2.1 Inspection des colonnes suspectes (rating, bedrooms, baths)

In [6]:
# Inspection des colonnes au type suspect
print("=== RATING — valeurs uniques (10 premières) ===")
print(df['rating'].value_counts().head(10))
print(f"\nNombre de valeurs uniques : {df['rating'].nunique()}")

print("\n=== BEDROOMS — valeurs uniques ===")
print(df['bedrooms'].value_counts().head(10))

print("\n=== BATHS — valeurs uniques ===")
print(df['baths'].value_counts().head(10))

=== RATING — valeurs uniques (10 premières) ===
rating
No rating    3593
5.0          2882
4.67          517
4.75          469
4.80          439
4.83          426
4.88          424
4.86          421
4.89          378
4.92          369
Name: count, dtype: int64

Nombre de valeurs uniques : 162

=== BEDROOMS — valeurs uniques ===
bedrooms
1         13439
2          3602
Studio     1815
3          1394
4           345
5           112
6            29
7             9
9             6
8             5
Name: count, dtype: int64

=== BATHS — valeurs uniques ===
baths
1                17015
2                 2082
1.5               1001
2.5                240
3                  171
0                   82
4                   69
3.5                 50
4.5                 18
Not specified       13
Name: count, dtype: int64


### 2.2 Plan de nettoyage

In [7]:
print("=== BATHS — valeurs uniques ===")
print(df['baths'].value_counts().head(15))

=== BATHS — valeurs uniques ===
baths
1                17015
2                 2082
1.5               1001
2.5                240
3                  171
0                   82
4                   69
3.5                 50
4.5                 18
Not specified       13
5                    6
6                    4
5.5                  2
6.5                  2
11.5                 1
Name: count, dtype: int64


### 2.3 — Conversion de la colonne `rating`
La colonne contient 3 593 fois la valeur textuelle `"No rating"` qui correspond aux logements n'ayant pas encore reçu de note. On les remplace par NaN (valeur manquante propre) puis on convertit la colonne en numérique.

In [8]:
# Remplacement de "No rating" par NaN
df['rating'] = df['rating'].replace('No rating', np.nan)

# Conversion en numérique (errors='coerce' transforme tout texte restant en NaN par sécurité)
df['rating'] = pd.to_numeric(df['rating'], errors='coerce')

# Vérification
print(f"Type de la colonne : {df['rating'].dtype}")
print(f"Valeurs manquantes : {df['rating'].isna().sum():,} ({df['rating'].isna().mean()*100:.1f}%)")
print(f"\nStatistiques :")
print(df['rating'].describe())

Type de la colonne : float64
Valeurs manquantes : 3,752 (18.1%)

Statistiques :
count   17006.00
mean        4.73
std         0.29
min         1.75
25%         4.64
50%         4.81
75%         4.93
max         5.00
Name: rating, dtype: float64


### 2.4 — Conversion de la colonne `bedrooms`
- `"Studio"` → 0 (un studio est, par définition, sans chambre séparée)
- `"Not specified"` (13 lignes) → NaN, puis suppression de ces lignes (négligeable sur 20 758)

In [9]:
# Remplacement des valeurs textuelles
df['bedrooms'] = df['bedrooms'].replace({'Studio': '0', 'Not specified': np.nan})

# Conversion en numérique
df['bedrooms'] = pd.to_numeric(df['bedrooms'], errors='coerce')

# Suppression des lignes où bedrooms est NaN (très peu de lignes)
df = df.dropna(subset=['bedrooms'])

print(f"Type de la colonne : {df['bedrooms'].dtype}")
print(f"Distribution :")
print(df['bedrooms'].value_counts().sort_index())

Type de la colonne : int64
Distribution :
bedrooms
0      1815
1     13439
2      3602
3      1394
4       345
5       112
6        29
7         9
8         5
9         6
14        1
15        1
Name: count, dtype: int64


### 2.5 — Conversion de la colonne `baths`
Même traitement que `bedrooms` : `"Not specified"` (13 lignes) → NaN puis suppression. À noter qu'on observe une valeur extrême (11.5 salles de bain) qui sera filtrée à l'étape suivante.

In [10]:
# Remplacement des valeurs textuelles
df['baths'] = df['baths'].replace('Not specified', np.nan)

# Conversion en numérique
df['baths'] = pd.to_numeric(df['baths'], errors='coerce')

# Suppression des lignes où baths est NaN
df = df.dropna(subset=['baths'])

print(f"Type de la colonne : {df['baths'].dtype}")
print(f"Distribution :")
print(df['baths'].value_counts().sort_index())

Type de la colonne : float64
Distribution :
baths
0.00        82
1.00     17015
1.50      1001
2.00      2082
2.50       240
3.00       171
3.50        50
4.00        69
4.50        18
5.00         6
5.50         2
6.00         4
6.50         2
7.00         1
11.50        1
15.50        1
Name: count, dtype: int64


### 2.6 — Suppression des outliers (valeurs aberrantes)
Justification des seuils :
- **`price`** ≤ 1 000 \$/nuit : couvre 99\% du marché new-yorkais, élimine les valeurs extrêmes ( max observé : 100 000 \$ ).
- **`minimum_nights`** ≤ 365 : élimine les locations longue durée déguisées (max observé : 1 250 nuits).
- **`beds`** ≤ 10 : conserve uniquement les vrais Airbnb, exclut les hôtels camouflés.
- **`baths`** ≤ 5 : élimine les erreurs de saisie (max observé : 11.5 salles de bain).

In [11]:
lignes_avant = len(df)

df = df[
    (df['price'] >= 10) & (df['price'] <= 1000) &
    (df['minimum_nights'] <= 365) &
    (df['beds'] <= 10) &
    (df['baths'] <= 6)
]

lignes_apres = len(df)
print(f"Lignes supprimées : {lignes_avant - lignes_apres:,}")
print(f"Lignes restantes : {lignes_apres:,} ({lignes_apres/lignes_avant*100:.1f}% conservé)")

Lignes supprimées : 195
Lignes restantes : 20,550 (99.1% conservé)


### 2.7 Vérification finale

In [12]:
# Reset de l'index après suppressions
df = df.reset_index(drop=True)

print(f"Dataset final : {df.shape[0]:,} lignes × {df.shape[1]} colonnes")
print(f"\nTypes des colonnes critiques :")
print(df[['price', 'rating', 'bedrooms', 'baths', 'beds', 'number_of_reviews']].dtypes)
print(f"\nStatistiques après nettoyage :")
df[['price', 'minimum_nights', 'beds', 'baths', 'rating']].describe()

Dataset final : 20,550 lignes × 22 colonnes

Types des colonnes critiques :
price                float64
rating               float64
bedrooms               int64
baths                float64
beds                   int64
number_of_reviews      int64
dtype: object

Statistiques après nettoyage :


,price,minimum_nights,beds,baths,rating
count,20550.00,20550.00,20550.00,20550.00,16847.00
mean,161.13,27.91,1.70,1.17,4.73
std,131.63,22.71,1.12,0.44,0.29
min,10.00,1.00,1.00,0.00,1.75
25%,80.00,30.00,1.00,1.00,4.64
50%,125.00,30.00,1.00,1.00,4.81
75%,199.00,30.00,2.00,1.00,4.93
max,1000.00,365.00,10.00,6.00,5.00


---
## Étape 3 — Enrichissement (Feature Engineering)
Création de nouvelles variables pertinentes pour l'analyse d'investissement :
1. **`revenu_estime`** : revenu mensuel estimé par logement
2. **`categorie_prix`** : segmentation Low-cost / Standard / Luxe
3. **`type_hote`** : Particulier vs Professionnel (selon le nombre de logements possédés)

### 3.1 — `revenu_estime` : revenu mensuel estimé
Formule : `revenu_estime = price × reviews_per_month`

Cette estimation repose sur l'hypothèse que chaque avis correspond à une réservation. C'est une approximation conservative (en réalité, seuls ~50% des voyageurs laissent un avis), mais elle reste utile pour **comparer les logements entre eux**.

In [13]:
# Création de la colonne revenu estimé mensuel
df['revenu_estime'] = df['price'] * df['reviews_per_month']

# Aperçu
print("Statistiques du revenu estimé mensuel :")
print(df['revenu_estime'].describe().round(2))

Statistiques du revenu estimé mensuel :
count   20550.00
mean      202.13
std       356.96
min         0.39
25%        24.00
50%        82.00
75%       249.09
max     10201.64
Name: revenu_estime, dtype: float64


### 3.2 — `categorie_prix` : segmentation du marché
Découpage en trois segments pour faciliter l'analyse :
- **Low-cost** : < 100 $/nuit
- **Standard** : 100 à 250 $/nuit
- **Luxe** : > 250 $/nuit

Ces seuils correspondent approximativement aux quartiles du marché new-yorkais.

In [14]:
# Création de la catégorie de prix avec pd.cut
bins = [0, 100, 250, df['price'].max() + 1]   # bornes des intervalles
labels = ['Low-cost', 'Standard', 'Luxe']

df['categorie_prix'] = pd.cut(df['price'], bins=bins, labels=labels, right=False)

# Vérification de la répartition
print("Répartition par catégorie de prix :")
print(df['categorie_prix'].value_counts())
print(f"\nEn pourcentage :")
print((df['categorie_prix'].value_counts(normalize=True) * 100).round(1).astype(str) + ' %')

Répartition par catégorie de prix :
categorie_prix
Standard    9984
Low-cost    7323
Luxe        3243
Name: count, dtype: int64

En pourcentage :
categorie_prix
Standard    48.6 %
Low-cost    35.6 %
Luxe        15.8 %
Name: proportion, dtype: str


### 3.3 — `type_hote` : Particulier vs Professionnel
Variable bonus qui distingue :
- **Particulier** : possède 1 logement (loue probablement sa résidence)
- **Multi-propriétaire** : 2 à 5 logements
- **Professionnel** : 6 logements ou plus (gestion semi-industrielle)

Cette segmentation est utile pour un fonds d'investissement : on observe différemment un marché dominé par des pros vs un marché de particuliers.

In [15]:
# Fonction de catégorisation
def categoriser_hote(n):
    if n == 1:
        return 'Particulier'
    elif n <= 5:
        return 'Multi-propriétaire'
    else:
        return 'Professionnel'

df['type_hote'] = df['calculated_host_listings_count'].apply(categoriser_hote)

# Répartition
print("Répartition par type d'hôte :")
print(df['type_hote'].value_counts())
print(f"\nEn pourcentage :")
print((df['type_hote'].value_counts(normalize=True) * 100).round(1).astype(str) + ' %')

Répartition par type d'hôte :
type_hote
Particulier           8956
Multi-propriétaire    6477
Professionnel         5117
Name: count, dtype: int64

En pourcentage :
type_hote
Particulier           43.6 %
Multi-propriétaire    31.5 %
Professionnel         24.9 %
Name: proportion, dtype: str


### 3.4 — Sauvegarde du dataset nettoyé et enrichi

In [16]:
# Export du dataset prêt pour l'analyse
df.to_csv('../data/processed/airbnb_nyc_clean.csv', index=False)
print(f"✅ Dataset sauvegardé : {df.shape[0]:,} lignes × {df.shape[1]} colonnes")
print(f"   Chemin : data/processed/airbnb_nyc_clean.csv")

✅ Dataset sauvegardé : 20,550 lignes × 25 colonnes
   Chemin : data/processed/airbnb_nyc_clean.csv
